src/os/bluestore/BlueStore.cc

```diff
@@ -6384,7 +6384,7 @@ int BlueStore::_open_db_and_around(bool read_only, bool to_repair)
   }
 
   // when function is called in repair mode (to_repair=true) we skip db->open()/create()
-  if (!read_only && !to_repair && cct->_conf->bluestore_allocation_from_file
+  if (!is_rotational() && !read_only && !to_repair && cct->_conf->bluestore_allocation_from_file
```

Comment:
seems like we want a new function BlueFS::is_db_rotational() -- just like the existing BlueFS::is_wal_rotational() -- which would check the db bdev, or if none, the block bdev in the same way.

BlueStore::is_rotational() is checking the main data block device afaict